purchases/src/main/kotlin/com/revenuecat/purchases/google/usecase/QueryProductDetailsUseCase.kt

```diff
@@ -114,4 +119,37 @@ internal class QueryProductDetailsUseCase(
             responseTime = Duration.between(requestStartTime, useCaseParams.dateProvider.now),
         )
     }
+
+    // This error was introduced in Google's Billing Client 7.1.0 and 7.1.1. Logging an error to notify the developer,
+    // as soon as possible that purchases will fail unless core library desugaring is enabled.
+    private companion object {
+        val hasLoggedBillingFlowParamsError = AtomicBoolean(false)
+    }
+
+    @Suppress("NestedBlockDepth")
+    private fun logErrorIfIssueBuildingBillingParams(productDetails: List<ProductDetails>) {
+        productDetails.firstOrNull()?.let {
+            if (hasLoggedBillingFlowParamsError.getAndSet(true)) {
+                return
+            }
+            val productDetailsParams = try {
+                val offerToken = it.subscriptionOfferDetails?.firstOrNull()?.offerToken
+                ProductDetailsParams.newBuilder()
+                    .setProductDetails(it)
+                    .apply { if (offerToken != null) setOfferToken(offerToken) }
+                    .build()
+            } catch (@Suppress("TooGenericExceptionCaught") e: Throwable) {
+                errorLog("Error building ProductDetailsParams during safety check.", e)
+                null
+            }
+
+            if (productDetailsParams == null) return
+
+            try {
+                BillingFlowParams.newBuilder().setProductDetailsParamsList(listOf(productDetailsParams)).build()
+            } catch (e: NoClassDefFoundError) {
+                errorLog(NO_CORE_LIBRARY_DESUGARING_ERROR_MESSAGE, e)
+            }
```

Comment:
In addition to `NoClassDefFoundError`, shouldn't we also catch here the more generic error just in case? I don't know if it makes sense, but to make sure that we never throw an error from this safety check method (i.e. same reason as above)